### DISCUSSION TITLE
Upgrade from 2.4.20 to 2.4.40 partailly failed

### DISCUSSION THREAD

**Discussion by UMWP** [Security-Onion-Solutions/securityonion#12311] [2.4]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

upgrading

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

8

### RAM

16

### Storage for /

240GB

### Storage for /nsm

1,8TB

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

Distributed on-prem deployment with 2 sensors, 1 search and 1 management node.

After soup from 2.4.20 to 2.4.40 so-elasticsearch status is missing on management node
All other nodes are OK on grid status.

sudo salt-call state.highstate on management shows 6 failed.

Logs included.

[soup.log](https://github.com/Security-Onion-Solutions/securityonion/files/14164770/soup.log)
[state.highstate.txt](https://github.com/Security-Onion-Solutions/securityonion/files/14164773/state.highstate.txt)

Any help appreciated

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by InfosecGoon** [comment]
Are there any useful errors in the Elasticsearch logs on the Manager Node? They're located in /opt/so/log/elasticsearch/securityonion.log.

**Comment by petiepooo** [comment]
Elasticsearch is not running.  In the soup log was:

> ID: so-elasticsearch
>    Function: docker_container.running
>      Result: False
>     Comment: One or more requisite failed: ssl./etc/pki/elasticsearch.crt, ca.pki_public_ca_crt, ssl./etc/pki/elasticsearch.key
>     Started: 09:11:11.382131
>    Duration: 0.006 ms
>     Changes:   

Well before that, your first false result was regarding the ca.crt, which is noted above in the comment as well.  I've seen a similar issue and ended up reinstalling as it was a lab system, but you may get by with recreating all your SSL certs.  I think the way to do that is `salt-call state.apply ssl.remove -l info` then restart everything by either rebooting each node or bouncing the docker service and running `so-checkin` on each node, starting with the manager.

**Comment by UMWP** [comment]
First of all thank You very much and I'm so sorry for my terrible mistake.
It's so-elastalert container with status "missing" (and not so-elasticsearch), I'm very sorry about that false info.
Lines from  so-elastalert-restart and from /opt/so/log/elastalert/elastalert.log containing errors are included below:

[so-elastalert restart.txt](https://github.com/Security-Onion-Solutions/securityonion/files/14191041/so-elastalert.restart.txt)

[Error.txt](https://github.com/Security-Onion-Solutions/securityonion/files/14190915/Error.txt)

And also Kibana shows "404 page not found"

Thank You very much once again.

**Reply by petiepooo** [reply]
No.  Elastalert is only failing because it cannot reach Elasticsearch.  Same with Kibana.  To see that, run `salt-call state.apply ssl -l info`, and you will see the certificate errors that are causing all the dependent containers to fail.  My recommendations stand.

**Comment by UMWP** [comment]
Unfortunately after:
salt-call state.apply ssl.remove -l info

and then after restarting all nodes (management, forwarding, search) it's still the same:
so-elastalert status "missing"

I think I'll have to revert back to 2.4.20 :-(

**Reply by petiepooo** [reply]
Sorry to hear that.  Good luck.
